### PR TITLE
[Auth] Test respond error

### LINKED ISSUE #459: [Auth] handle error response

### PR DESCRIPTION
**Pull request by llbartekll**
# Description
- handle peer error response 
- add testUserRespondError
- add reject method to public interface

Resolves #459 #460

## How Has This Been Tested?
- integration tests

## Due Dilligence

* [ ] Breaking change
* [ ] Requires a documentation update

### CODE REVIEW

File: Sources/Auth/AuthClient.swift [hunk 1]
```diff
@@ -87,9 +88,13 @@ public class AuthClient {
...
     }
 
-    public func respond(requestId: RPCID, result: Result<CacaoSignature, Never>) async throws {
+    public func respond(requestId: RPCID, signature: CacaoSignature) async throws {
         guard let account = account else { throw Errors.unknownWalletAddress }
-        try await walletRespondService.respond(requestId: requestId, result: result, account: account)
+        try await walletRespondService.respond(requestId: requestId, signature: signature, account: account)
+    }
+
+    public func reject(requestId: RPCID) async throws {
```
Comment (flypaper0) [inline Sources/Auth/AuthClient.swift:96]: Good job!

File: Sources/Auth/Services/App/AppRespondSubscriber.swift [hunk 2]
```diff
@@ -28,16 +28,23 @@ class AppRespondSubscriber {
...
             guard
-                let requestId = subscriptionPayload.response.id,
+                let requestId = response.id,
                 let request = rpcHistory.get(recordId: requestId)?.request,
                 let requestParams = request.params, request.method == "wc_authRequest"
             else { return }
 
             networkingInteractor.unsubscribe(topic: subscriptionPayload.topic)
 
+            if let errorResponse = response.error,
```
Comment (flypaper0) [inline Sources/Auth/Services/App/AppRespondSubscriber.swift:40]: guard?

**Review by flypaper0 — APPROVED** [review]

File: Sources/Auth/Services/App/AppRespondSubscriber.swift (hunk 2 above)
Comment (llbartekll) [reply Sources/Auth/Services/App/AppRespondSubscriber.swift:40]: if we have guard and response is not error the function would return here but we want to handle error and success response right?

Comment (llbartekll) [reply Sources/Auth/Services/App/AppRespondSubscriber.swift:40]: > We sending error code in reject method, and recover AuthError with code here. Right?

correct

**Pull request merged**